Sources/StreamChat/ChatClient.swift

```diff
@@ -243,7 +243,7 @@ public class ChatClient {
     /// The default configuration of URLSession to be used for both the `APIClient` and `WebSocketClient`. It contains all
     /// required header auth parameters to make a successful request.
     private var urlSessionConfiguration: URLSessionConfiguration {
-        let configuration = URLSessionConfiguration.default
+        let configuration = config.urlSessionConfiguration
         configuration.waitsForConnectivity = false
         configuration.httpAdditionalHeaders = sessionHeaders
         configuration.timeoutIntervalForRequest = config.timeoutIntervalForRequest
```

Comment:
One downside here is that customers won't be able to override the `waitsForConnectivity` or the `httpAdditionalHeaders`. But for now, I think this approach is simple enough, and it should work for 99.99% of use cases. If we see the need, we can add this feature later 👍